credentialdigger/cli/get_discoveries.py

```diff
@@ -0,0 +1,279 @@
+        except OSError as osE:
+            console.print(f'[red]{osE}[/]')
+
+
+def assign_categories(client, discoveries):
+    """ Add category to each discovery
+
+    Parameters
+    ----------
+    client: `credentialdigger.Client`
+        Instance of the client from which we retrieve rules
+    discoveries: list
+        List of discoveries without assigned categories to them
+    """
+    rulesdict = client.get_rules()
+    for discovery in discoveries:
+        if discovery['rule_id']:
+            category = rulesdict[discovery['rule_id'] - 1]['category']
+            discovery['category'] = category
+        else:
+            discovery['category'] = '(rule deleted)'
+
+
+def filter_discoveries(discoveries, state):
+    """ Filter discoveries based on state
+
+    Parameters
+    ----------
+    discoveries: list
+        List of discoveries to be filtered
+    states: str | list
+            - str: if it equals 'all', then return all discoveries.
+                   return chosen state otherwise (i.e 'false_positive')
+            - list: return all the discoveries that have states contained
+                    in this list (i.e ['new', 'false_positive'])
+
+    Returns
+    -------
+    list
+        Filtered list of discoveries
+    """
+    if state == 'all':
+        state = ['new', 'false_positive',
+                 'addressing', 'not_relevant', 'fixed']
+
+    filtered_discoveries = list(
+        filter(lambda d: d.get('state') in state, discoveries))
+
+    return filtered_discoveries
+
+
+def run(client, args):
+    """
+    Retrieve discoveries of a git repository and export them if needed
+
+    Parameters
+    ----------
+    client: `credentialdigger.Client`
+        Instance of the client from which we retrieve results
+    args: `argparse.Namespace`
+        Arguments from command line parser.
+    """
+
+    try:
+        discoveries = client.get_discoveries(
+            repo_url=args.repo_url, file_name=args.filename)
+    except Exception as e:
+        console.print(f'[red]{e}[/]')
+
+    # If --state is specified, filter the discoveries based on it
+    if args.state is not None:
+        discoveries = filter_discoveries(discoveries, args.state)
+
+    # if --save is specified, export the discoveries and exit
+    if args.save is not None:
+        export_csv(discoveries, client, save=args.save)
+        sys.exit(len(discoveries))
+
+    if len(discoveries) == 0:
+        # if repo has no discoveries, exit
+        console.print(f'[bold] {args.repo_url} has 0 discoveries.')
+    elif len(discoveries) > MAX_NUMBER_DISCOVERIES:
+        response = ''
+        while response.upper() not in ['Y', 'N', 'YES', 'NO']:
+            response = console.input(
+                f'[bold]This repository has more than {MAX_NUMBER_DISCOVERIES} \
+discoveries, export them as .csv instead? (Y/N) ')
+        if response.upper() in ['N', 'NO']:
+            print_discoveries(discoveries, args.repo_url)
+        else:
```

Comment:
so if someone types 'Z' shall we set it to "yes"?

Reply:
No, if someone types 'Z', he will be asked to enter another input in ['Y', 'N', 'YES', 'NO'] and will not be able to leave the **while** loop.

When testing `if response.upper() in ['N', 'NO']:`, we know that the input will 100% be in the list ['Y', 'N', 'YES', 'NO']. so if it is not 'N' nor 'NO', then it'll be 'Y' or 'YES'.